### DISCUSSION TITLE
How to run examples/http3_server on different host address

### DISCUSSION THREAD

**Discussion by ar-ag** [aiortc/aioquic#512] [Q&A]
I am trying to run examples/http3_server on a host in mininet, with the address `10.0.0.1`

`python3 /home/ar_ag/mininet/aioquic/examples/http3_server.py --certificate /home/ar_ag/mininet/ssl_cert.pem --private-key /home/ar_ag/mininet/ssl_key.pem --host 10.0.0.1`

But when I run the client script at different host in mininet, with address `10.0.0.2`

`python3 /home/ar_ag/QUICforge/aioquic/examples/http3_client.py --ca-certs /home/ar_ag/mininet/pycacert.pem https://10.0.0.1:4433/`

the connection is terminating after the handshake, with the following warning

```
2024-06-19 15:31:50,107 WARNING quic [5f128ac9e255c158] Error: 298, reason: hostname '10.0.0.1' doesn't match DNSPattern(pattern=b'localhost'), frame_type: 6
2024-06-19 15:31:50,107 INFO quic [5f128ac9e255c158] Connection close sent (code 0x12A, reason hostname '10.0.0.1' doesn't match DNSPattern(pattern=b'localhost'))
```

**Comment by YifangZhang** [comment]
Since Aioquic only allows the H3 connection, which the example only allows an URL, you can define a pre-defined URL at your "/etc/hosts". For example, you can define as "10.0.0.1 www.test.com". Then you can have:

python3 /home/ar_ag/QUICforge/aioquic/examples/http3_client.py --ca-certs /home/ar_ag/mininet/pycacert.pem www.test.com

Hope it helps!
